apps/music/test/marionette/loading_test.js

```diff
@@ -0,0 +1,76 @@
+/* global require, marionette, setup, suite, test, __dirname */
+'use strict';
+
+var assert = require('assert');
+var Music = require('./lib/music.js');
+var FakeRingtones = require('./lib/fakeringtones.js');
+var FakeControls = require('./lib/fakecontrols.js');
+
+marionette('Music loading', function() {
+  var apps = {};
+  apps[FakeRingtones.DEFAULT_ORIGIN] = __dirname + '/fakeringtones';
+  apps[FakeControls.DEFAULT_ORIGIN] = __dirname + '/fakecontrols';
+
+  var client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var music;
+
+  setup(function() {
+    music = new Music(client);
+
+  });
+
+  suite('Detection of movies.', function () {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        },
+        {
+          type: 'music',
+          filePath: 'test_media/Movies/gizmo2.mp4'
+        }
```

Comment:
In general, this is a hard test to do exhaustively because mp4 is such a generic container and there are so many codecs that get included within it. Ideally, we'd have a comprehensive library of mp4 audio and video samples and could write a test to verify that the app reads all of the audio files and none of the video files.  If you know of any such mp4 sample library, feel free to import a copy into test-data/mp4/ or something and expand this test.  If not, then just make sure that the test is written to be extensible so that we can easily add other .mp4 files to it in the future.

Speaking of a test-data/mp4/ directory. Consider creating one, and copying the gizmo file and test-data/*.m4a file into it so that there is one central location for the test media used here.